### PR TITLE
Complex featurizer fix

### PR DESCRIPTION
**Pull request by ncfrey**
This PR refactors `dc.feat.base_classes.ComplexFeaturizer` to inherit from `Featurizer`. It should standardize complex featurizers to behave as expected for featurizers in DeepChem. I also added some missing tests for complex featurizers.

### CODE REVIEW

File: deepchem/feat/base_classes.py [hunk 1]
```diff
@@ -150,22 +150,20 @@ def __str__(self) -> str:
...
     Calculate features for mol/protein complexes.
 
     Parameters
     ----------
-    mols: List[str]
-      List of PDB filenames for molecules.
-    protein_pdbs: List[str]
-      List of PDB filenames for proteins.
+    complexes: Iterable[Tuple[str, str]]
+      List of filenames (PDB, SDF, etc.) for ligand molecules and proteins.
```
Comment (peastman) [inline deepchem/feat/base_classes.py:166]: The description is a little ambiguous.  How about adding another sentence: "Each element should be a tuple of the form (ligand_filename, protein_filename)."

**Comment by peastman** [comment]
This looks a lot cleaner than before!

**Review by rbharath — APPROVED** [review]
This looks quite nice now!

**Comment by ncfrey** [comment]
Awesome! I fixed a few fingerprint tests and expanded the docs as @peastman suggested.

**Comment by rbharath** [comment]
Looks good on my end!

**Comment by peastman** [comment]
Looks good to merge once the CI errors are fixed (aside from the test_tokenize errors that aren't related to this PR).  flake8 is complaining about a few things.

**Comment by ncfrey** [comment]
Great, I fixed the flake8 errors and will merge in once the checks are done.

**Comment by ncfrey** [comment]
Failures are unrelated to this PR so I'm going to merge it in.

**Pull request merged**